lib/chat_transcript_service.rb

```diff
@@ -60,9 +60,15 @@ def channel_attr
     end
   end
 
-  def initialize(channel, message_ids)
+  def initialize(channel, message_ids: [], messages: [])
```

Comment:
I have mixed feeling about that. I love how it is cached so when messages are passed then no additional queries are sent to the database. 

The thing I don't like is that if I want to use that class quickly and just read initializer without details, I may use it as:
`ChatTranscriptService.new("channel", messages: [Message.last], message_ids: [1, 2, 3])`

I have seen before helper initializers to avoid that problem, something like:
```ruby
private_class_method :new
def self.create_with_message_ids(channel, ids)
  new(channel, message_ids: ids)
end
def self.create_with_messages(channel, messages)
  new(channel, messages: messages)
end
```

Not sure if that would be better, so I am leaving decision to you :)

Reply:
Yeah fair enough...another way I could do it is like this:

```ruby
def initializer(channel, messages_or_ids:)
  if messages_or_ids.all? {|m| m.is_a? Numeric}
    @message_ids = messages_or_ids
  else
    @messages = messages_or_ids
  end
end
```

Might make that change actually, it is a bit neater.